src/lib/sort/sort-header-intl.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable} from '@angular/core';
+import {SortDirection} from './sort-direction';
+
+/**
+ * To modify the labels and text displayed, create a new instance of MdSortHeaderIntl and
+ * include it in a custom provider.
+ */
+@Injectable()
+export class MdSortHeaderIntl {
+  sortButtonLabel = (id: string) => {
+    return `Change sorting for ${id}`;
+  }
+
+  /** A label to describe the current sort (visible only to screenreaders). */
+  sortDescriptionLabel = (id: string, direction: SortDirection) => {
+    return `Sorted by ${id} ${direction}`;
```

Comment:
If direction is 'asc' | 'desc', won't that sound strange with a screenreader? I wonder if you want to override it to be a real word, but I'm not super familiar with what the standards are here.